common.go

```diff
@@ -40,6 +36,10 @@ func NullString() *string {
 	return output
 }
 
+func NewEnum[T ~string](value T) *T {
```

Comment:
Here `~string` means that `T` must have an underlying type that is a string.
Golang ref [here](https://go.dev/blog/deconstructing-type-parameters)